utils.py

```diff
@@ -103,6 +171,52 @@ def green(string):
     print(f"\n\033[01;32m{string}\033[0m", flush=True)
 
 
+def prepare_initrd(architecture, rootfs_format='cpio'):
+    """
+    Returns a decompressed initial ramdisk.
+
+    Parameters:
+        architecture (str): Architecture to download image for.
+        rootfs_format (str): Initrd format ('cpio' or 'ext4')
+    """
+    src = Path(BOOT_UTILS, 'images', architecture,
+               f"rootfs.{rootfs_format}.zst")
+    src.parent.mkdir(exist_ok=True, parents=True)
+
+    # First, make sure that the current user is not rate limited by GitHub,
+    # otherwise the next API call will not return valid information.
+    gh_json_rl = get_gh_json('https://api.github.com/rate_limit')
+    limit = gh_json_rl['resources']['core']['limit']
+    remaining = gh_json_rl['resources']['core']['remaining']
+
+    # If we have API calls remaining, we can query for the latest release to
+    # make sure that we are up to date.
+    if remaining > 0:
+        gh_json_rel = get_gh_json(
+            f"https://api.github.com/repos/{REPO}/releases/latest")
+        # Download the ramdisk if it is not already downloaded
+        if not src.exists():
+            download_initrd(gh_json_rel, src)
+        # If it is already downloaded, check that it is up to date and download
+        # an update only if necessary.
+        elif (rel_file := src.with_name('.release')).exists():
+            cur_rel = rel_file.read_text(encoding='utf-8')
+            latest_rel = gh_json_rel['tag_name']
+            if cur_rel != latest_rel:
+                download_initrd(gh_json_rel, src)
+    elif not src.exists():
```

Comment:
This probably deserves a better comment but I wrote it this way so that if there are no more queries available under the current rate limit and the rootfs has already been downloaded, the script skips attempting to query the API for the latest release (which would fail). I could add a warning for that situation ("Hit GitHub rate limit for querying latest release, your downloaded image may be out of date"?), which might make things a little clearer as well.